dart/lib/src/sentry_tracer.dart

```diff
@@ -41,7 +46,23 @@ class SentryTracer extends ISentrySpan {
     if (!_rootSpan.finished &&
         (!_waitForChildren || _haveAllChildrenFinished())) {
       _rootSpan.status ??= status;
-      await _rootSpan.finish();
+
+      var _rootEndTimestamp = getUtcDateTime();
+      if (_trimEnd && children.isNotEmpty) {
+        final childEndTimestamps = children
+            .where((child) => child.endTimestamp != null)
+            .map((child) => child.endTimestamp!);
+
+        if (childEndTimestamps.isNotEmpty) {
+          final oldestChildEndTimestamp =
+              childEndTimestamps.reduce((a, b) => a.isAfter(b) ? a : b);
+          if (_rootEndTimestamp.isAfter(oldestChildEndTimestamp)) {
+            _rootEndTimestamp = oldestChildEndTimestamp;
+          }
+        }
+      }
```

Comment:
Should this be maybe after `finish unfinished spans otherwise transaction gets dropped`? otherwise it could lead to longer spans than transactions by a few millis (off-by-1 problem)